test/katex-spec.js

```diff
@@ -2330,3 +2336,40 @@ describe("The maxSize setting", function() {
         expect(built.style.borderTopWidth).toEqual("0em");
     });
 });
+
+describe("Tree attributes propagation", function() {
+    it("should put tree attributes to DOM nodes", function() {
+        const expr = "\\frac 1 2";
+        const tree = getParsed(expr);
+
+        tree[0].attributes = {
+            "katex-frac-id": "frac",
+        };
+
+        tree[0].value.numer.attributes = {
+            "katex-numer-id": "numer",
+        };
+
+        tree[0].value.denom.attributes = {
+            "katex-denom-id": "denom",
+        };
```

Comment:
You're going to want to add `id`s to each node automatically eventually though.  I wonder if provide some sort of callback that is passed each parse tree node and its associated HTML tree node might be better.  The callback would be called whenever a method on groupTypes in buildHTML.js returns.

Reply:
I moved attributes logic into post-processing funciton. Currently it is set to Options.postProcessor, which makes Options non-serializable. Should it be serialzable? If yes, I can move post-processing into some wrapper around Options. Or into signatures of every method of buildHTML.js, but I guess it would be a bit dirty.

Simple numeric IDs are currently assigned inside test. I can make an option to set them automatically.